Source/Core/Core/IOS/Network/KD/VFF/VFFUtil.cpp

```diff
@@ -0,0 +1,297 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/IOS/Network/KD/VFF/VFFUtil.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <mutex>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <fmt/format.h>
+
+// Does not compile if diskio.h is included first.
+// clang-format off
+#include "ff.h"
+#include "diskio.h"
+// clang-format on
+
+#include "Common/Align.h"
+#include "Common/FatFsUtil.h"
+#include "Common/FileUtil.h"
+#include "Common/IOFile.h"
+#include "Common/Logging/Log.h"
+#include "Common/ScopeGuard.h"
+#include "Common/StringUtil.h"
+#include "Common/Swap.h"
+
+#include "Core/IOS/Uids.h"
+
+static IOS::HLE::FS::FileHandle* s_vff;
```

Comment:
This can now be a member of the `VffFatFsCallbacks` class and be passed to the individual vff functions.